lumispy/signals/luminescence_spectrum.py

```diff
@@ -378,20 +378,21 @@ def px_to_nm_grating_solver(self, gamma_deg, deviation_angle_deg, focal_length_m
                                          gamma_deg, deviation_angle_deg, focal_length_mm, ccd_width_mm,
                                          grating_central_wavelength_nm, grating_density_gr_mm)
 
-        # in place conversion
         if inplace:
-            self.axes_manager.remove(-1)
-            self.axes_manager._axes.append(nm_axis)
-            return
-
-        # create and return new signal
+            s = self
         else:
             s = self.deepcopy()
-            s.axes_manager.remove(-1)
-            s.axes_manager._axes.append(nm_axis)
+
+        s.axes_manager.remove(-1)
+        s.axes_manager._axes.append(nm_axis)
+
+        if inplace:
+            return
+        else:
             return s
```

Comment:
`return s` is sufficient in this case, as it is a pointer to self in case of `inplace`